src/planscape/planning/management/commands/performance_report.py

```diff
@@ -0,0 +1,128 @@
+import os
+from datetime import date
+from django.core.mail import send_mail
+from django.core.management.base import BaseCommand, CommandParser
+from django.db import connection
+from planscape import settings
+from utils.string_utils import snake_to_capitals
+
+
+class Command(BaseCommand):
+    help = "Runs some SQL reports about performance and either emails them or outputs to stdout"
+
+    message_text = ""
+    emailing = False
+    recipient = None
+
+    query_folder = "planning/report_queries/"
+
+    def process_queries(self):
+        query_path = os.path.join(settings.BASE_DIR, self.query_folder)
+        files = os.listdir(query_path)
+        for f in files:
+            self.output_report(f)
+
+    def output_report(self, query_file):
+        title = snake_to_capitals(query_file)
+        query_path = os.path.join(settings.BASE_DIR, self.query_folder, query_file)
+        # Check if query file exists
+        if not os.path.isfile(query_path):
+            self.stderr.write(f"Error: File not found: {query_path}")
+            return
+        try:
+            with open(query_path, "r", encoding="UTF-8") as f:
+                sql = f.read()
+        except OSError as e:
+            self.stderr.write(f"Error opening file: {e}")
+
+        self.write(f"\n\n{title}\n")
+
+        with connection.cursor() as cursor:
+            try:
+                cursor.execute(sql)
+                rows = cursor.fetchall()
+                column_descriptions = cursor.description
+
+                if rows:
+                    # Get header names and maximum widths
+                    headers = [desc[0] for desc in column_descriptions]
+                    max_widths = [
+                        max(len(str(row[i])) for row in rows)
+                        for i in range(len(headers))
+                    ]
+
+                    for i, header in enumerate(headers):
+                        max_widths[i] = max(max_widths[i] or 0, len(header))
+
+                    # Format headers and rows based on max widths found
+                    formatted_headers = "  ".join(
+                        header.ljust(width)
+                        for header, width in zip(headers, max_widths)
+                    )
+                    self.write(formatted_headers)
+
+                    for row in rows:
+                        formatted_row = "  ".join(
+                            str(value).ljust(width)
+                            for value, width in zip(row, max_widths)
+                        )
+                        self.write(formatted_row)
+                else:
+                    self.write("** No rows **")
+
+            except Exception as e:
+                self.stderr.write("Error executing SQL: {}".format(e))
```

Comment:
Looks like we are not capturing any of the exeption errors while sending the email (they are going out on stderr).
Not sure what the email will look in this case. Just empty? Do we wanna catch that somehow, just saying like "there was an error, etc et " on the email vs empty ?

Reply:
That's a great point. I added some output that will get included to the email.